src/cloud/sendCode.js

```diff
@@ -49,6 +56,16 @@ const sendCode = async request => {
     user.set('verificationStatus', status);
     user.set('verificationValid', valid);
     await user.save(null, { useMasterKey: true });
+
+    // set reservation as claimed and create a connection between users
+    if (reservation) {
```

Comment:
Lets move the updating of the reservation to the validate code func. sendCode will just check to make sure the reservation exists and is not claimed. Ill pass up the reservationId in validate code as well.